proxy/proxy.go

```diff
@@ -543,7 +543,7 @@ func mapRequest(ctx *context, requestContext stdlibcontext.Context, removeHopHea
 		rr = forwardToProxy(r, rr)
 	}
 
-	return rr, endpoint, nil
+	return rr, u.Host, nil
```

Comment:
Here `endpoint` was not nil only for loadbalanced endpoint and it is only used to track inflight requests. Now we want to have metrics key for network and loadbalanced backends but I am not sure we want to have it for dynamic backend (at least now).

If we want to have it for dynamic backend we'd need to normalize it as well.

Reply:
At the moment endpointregistry data essentially influences only the way how Skipper's load balancing algorithm chooses the backend address to send request to, the data for LBBackend could be modified from LBBackend or NetworkBackend. So, probably, it does not make much sense to normalize dynamic backends' hosts.

Firstly, it will create some additional and not straightforward changes.
Secondly, it will increase CPU and/or memory used in the hotpath.

Therefore I better retrieve metrics only for LBBackend and NetworkBackend.

Reply:
I would also postpone dynamic backend